### DISCUSSION TITLE
WireFrame Suggestion

### DISCUSSION THREAD

**Discussion by Saibaba-Alapati** [sugarlabs/musicblocks-v4#29] [Ideas]
MockDesign.

The work area will have its Scroll area, and I haven't included assuming that we can use touch actions or mouse pad actions.

While making this design my priority, it should handle both mobile and desktop with the help of breakpoints, so this type of design, the blocks, and other components will not take space horizontally. Hence, the work area is visible even on mobile, unlike the previous version of music blocks, which will mostly cover all the workspace areas. This design will be suitable for large, medium, and small size screens.

If this design is good and can be implemented, I can help implement it using React typescript.

Figma - https://www.figma.com/file/lbVRICZsYb8XPZGAcvKq7s/MusicBlocks?node-id=0%3A1 

 @meganindya  @walterbender Please share your comments, thougths, and helpful design changes.
 
Desktop and medium-size screens:-

Hidden:-

<img width="500" alt="Screenshot 2021-02-21 at 8 14 45 PM" src="https://user-images.githubusercontent.com/75042784/108628518-72c44280-7481-11eb-8fbc-16ea0aba6d29.png">

Expanded blocks:-  No horizontal Scroll Only Vertical
<img width="500" alt="Screenshot 2021-02-21 at 8 16 06 PM" src="https://user-images.githubusercontent.com/75042784/108628546-a2734a80-7481-11eb-8dfc-b0d7e9b0c9d4.png">

More Expanded:- No horizontal Scroll Only Vertical

<img width="500" alt="Screenshot 2021-02-21 at 8 16 43 PM" src="https://user-images.githubusercontent.com/75042784/108628559-b8810b00-7481-11eb-87c5-cdffa3c1be49.png">

This Action bar will have a horizontal scroll for medium size screens(like tabs)-

<img width="500" alt="Screenshot 2021-02-21 at 8 17 24 PM" src="https://user-images.githubusercontent.com/75042784/108628580-d2225280-7481-11eb-859a-011f6ee28536.png">

Mobile:-

   Hidden-
<img width="250" alt="Screenshot 2021-02-21 at 8 18 47 PM" src="https://user-images.githubusercontent.com/75042784/108628615-03028780-7482-11eb-8fb9-168e38cb9fda.png">

Expanded:- There is Horizontal scrolling but not vertical.

<img width="250" alt="Screenshot 2021-02-21 at 8 19 26 PM" src="https://user-images.githubusercontent.com/75042784/108628633-1a417500-7482-11eb-9b6c-b1b66e042299.png">

<img width="250" alt="Screenshot 2021-02-21 at 8 19 44 PM" src="https://user-images.githubusercontent.com/75042784/108628647-24fc0a00-7482-11eb-87af-4b1ee363ccf3.png">
  
  More expanded- Has vertical but not horizontal.
  
  
<img width="250" alt="Screenshot 2021-02-21 at 8 20 49 PM" src="https://user-images.githubusercontent.com/75042784/108628678-4b21aa00-7482-11eb-9700-e97b86ff5e3c.png">

<img width="250" alt="Screenshot 2021-02-21 at 8 21 06 PM" src="https://user-images.githubusercontent.com/75042784/108628691-55dc3f00-7482-11eb-82f8-bdb6dc54ff99.png">

Finally Search-

<img width="382" alt="Screenshot 2021-02-21 at 7 29 42 PM" src="https://user-images.githubusercontent.com/75042784/108627362-270e9a80-747b-11eb-9702-c93edd7df701.png">

Comparing with previous: -

Desktop:-
<img width="500" alt="Screenshot 2021-02-21 at 8 22 46 PM" src="https://user-images.githubusercontent.com/75042784/108628750-920f9f80-7482-11eb-9368-669fc93a50e5.png">
  
 Mobile:-
![prev](https://user-images.githubusercontent.com/75042784/108628835-f16daf80-7482-11eb-802e-330b9dcfa6a3.jpeg)

**Comment by walterbender** [comment]
We should take some inspiration from "Portfolio", Very simple, clean, compact.

https://flathub.org/apps/details/dev.tchx84.Portfolio

**Reply by Saibaba-Alapati** [reply]
sure I will check that out.

**Reply by Saibaba-Alapati** [reply]
> We should take some inspiration from "Portfolio", Very simple, clean, compact.
> 
> https://flathub.org/apps/details/dev.tchx84.Portfolio

Hey, Walter, I saw the portfolio I get that we need a very simple clean, and compact design. The main problem is we have too many subcategories in the application. we can achieve a simple design for the web as we have a good amount of space but on mobile, we have to conserve space I been actively looking just going through my ideas a good idea for mobile hasn't struck me I think we have to do separately for mobile with a different approach from the web. For now, if we going through the web I think we can do some different and good-looking design. Also, I wanted to know your thoughts on tailwind CSS over materials cause we can create custom designs using Tailwind CSS, and the materials mounting speed very low compared to tailwind CSS. If you have some other inspirations that suites our use case cause( too many subcategories) drop those links here I will check them out.

**Comment by walterbender** [comment]
I was thinking Portfolio for inspiration around palettes -- both for mobile and browsers. But in general, I think we need to diverge the UX between the two worlds. But the good news is that we have very good separation in the code.

Re tailwind, I have never used it but it looks pretty cool.

**Reply by Saibaba-Alapati** [reply]
Can we have its own app for music blocks ?

**Reply by Saibaba-Alapati** [reply]
@meganindya @walterbender  thoughts?

**Reply by meganindya** [reply]
Not sure I understand what you meant

**Reply by Saibaba-Alapati** [reply]
I think having a separate mobile app alongside with the website would be better and the advantages of having a separate mobile app would be :

- App can have their own offline and online capacities
- More user retention
- Instant updates and Notifications
- More convenient to the users
- Branding Possibilities

And I think working towards a separate Mobile app built using React Native would be more beneficial for us.

**Reply by Saibaba-Alapati** [reply]
I think it will be great if we are able musicblocks offline  @meganindya

**Reply by meganindya** [reply]
That was the idea. I personally believe an application like MB needs separate design and interaction languages for a touch based small screen device and a pointer based large screen device. Hence, I am not particularly interested in a responsive design to adapt to the significant size changes.

The architecture of the new app is such that the core state management is totally separate from the UI. So, we can write completely separate UI relatively quickly while using the same core. I'd like to explore the idea of a progressive web app before a native app because it'll be easier to implement. Eventually, we could explore native wrapping frameworks like react native.

But the web app needs to be built first.
